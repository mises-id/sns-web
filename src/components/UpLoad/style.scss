.grid-rows{
  @include m-grid($width:96vw,$rowSize:4);
}
.upload-box-size{
  width: calc(25vw - 17px);
  height: calc(25vw - 17px);
  margin-bottom: 15px;
  &.upload-box{
    background-color: #F8F8F8;
  }
  .file{
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    opacity: 0;
    width: calc(25vw - 17px);
    z-index: 33;
  }
  img{
    width: 100%;
    height: 100%;
  }
  .remove-image{
    width: 20px;
    height: 20px;
    position: absolute;
    right: -8px;
    top: -5px;
  }
}
.edit-image-box{
  height: 100vh;
}
.img-object-fit{
  object-fit: cover;
}
.camera-icon{
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%,-50%);
}