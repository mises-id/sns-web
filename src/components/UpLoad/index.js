/*
 * @Author: lmk
 * @Date: 2021-12-02 14:02:45
 * @LastEditTime: 2022-01-04 14:48:34
 * @LastEditors: lmk
 * @Description:
 */
import "@/styles/common.scss";
import "./style.scss";
import React, { useEffect, useState } from "react";
import { Toast, Popup } from "zarm";
import { useTranslation } from "react-i18next";
import removeImage from "@/images/deleteImage.png";
import cameraIcon from "@/images/camera.png";
import "react-photo-view/dist/index.css";
import EditImage from "../EditImage";
import Image from "../Image";

const Upload = ({imageList, setImageList}) => {
  const [maxLength, setmaxLength] = useState(9);
  const [activityImage, setactivityImage] = useState("");
  const [activityIndex, setactivityIndex] = useState(null);
  const [visible, setvisible] = useState(false);
  const { t } = useTranslation();
  // get change file
  const inputChange = (e) => {
    if (maxLength === 0) {
      Toast.show(t("upLoadFileWarning"));
      return false;
    }
    const files = Array.from(e.target.files)
      .slice(0, maxLength)
      .map((val) => {
        return {
          url:window.URL.createObjectURL(val),
          file:val
        }
      });
    console.log(files);
    const maxLen = maxLength - files.length || 0;
    setmaxLength(maxLen);
    imageList = [...imageList, ...files];
    setImageList(imageList);
  };
  // remove image item
  const deleteImage = (index) => {
    console.log(index)
    imageList.splice(index,1)
    setImageList(imageList);
    const len = 9 - imageList.length
    setmaxLength(len);
  };
  // preview image list
  const editImage = (index) => {
    const image = imageList[index];
    setactivityImage(image.url);
    setactivityIndex(index);
    setvisible(true);
  };
  const send = (image,index) => {
    imageList[index] = image;
    setImageList(imageList);
    closePop()
  };
  const closePop = () => {
    setactivityImage("");
    setactivityIndex(null);
    setvisible(false);
  };
  useEffect(() => {
    // setImageList(imageList)
    const maxLen = 9 - imageList.length;
    // console.log(maxLen)
    setmaxLength(maxLen);
  }, [imageList]) // eslint-disable-line react-hooks/exhaustive-deps
  return (
    <div className={`m-grid grid-rows m-padding8`}>
      {/* upload image list */}
      {imageList.map((item, index) => (
        <div className={`upload-box-size m-position-relative`} key={index}>
          <img
            src={removeImage}
            alt=""
            className="remove-image"
            onClick={() => deleteImage(index)}
          />
          <img
            src={item.url}
            alt=""
            className={`border-radius5 img-object-fit`}
            onClick={() => editImage(index)}
          />
        </div>
      ))}
      {/* upload components */}
      {maxLength > 0 && (
        <div
          className={`m-position-relative border-radius5 upload-box upload-box-size`}
        >
          <input
            type="file"
            className="file"
            onChange={inputChange}
            multiple
            maxLength={maxLength}
            accept="image/png,image/jpg,image/jpeg"
          />
          <div className="camera-icon">
            <Image size={25} source={cameraIcon} shape="square" height="auto"></Image>
          </div>
        </div>
      )}
      {/* image editor components */}
      <Popup
        visible={visible}
        direction="bottom"
        animationType="slideDown"
        mask={false}
      >
        {activityImage && (
          <EditImage
            image={activityImage}
            index={activityIndex}
            closePop={closePop}
            send={send}
          ></EditImage>
        )}
      </Popup>
    </div>
  );
};
export default Upload;
