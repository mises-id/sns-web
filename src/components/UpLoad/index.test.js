/*
 * @Author: lmk
 * @Date: 2021-07-15 10:12:21
 * @LastEditTime: 2022-01-04 16:43:12
 * @LastEditors: lmk
 * @Description: image test
 */
import React from 'react';
import {shallow} from 'enzyme';
import UpLoad from '.';
describe('UpLoad', () => {
  it('should render correctly', () => {
    const setImageList = ()=>([])
    const component = shallow(<UpLoad imageList={[]} setImageList={setImageList}/>)
    expect(component).toMatchSnapshot();
  });
});