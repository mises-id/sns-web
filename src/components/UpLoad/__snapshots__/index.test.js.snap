// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpLoad should render correctly 1`] = `
<div
  className="m-grid grid-rows m-padding8"
>
  <div
    className="m-position-relative border-radius5 upload-box upload-box-size"
  >
    <input
      accept="image/png,image/jpg,image/jpeg"
      className="file"
      maxLength={9}
      multiple={true}
      onChange={[Function]}
      type="file"
    />
    <div
      className="camera-icon"
    >
      <Image
        height="auto"
        shape="square"
        size={25}
        source="camera.png"
      />
    </div>
  </div>
  <ForwardRef(ForwardedRefComp)
    animationType="slideDown"
    direction="bottom"
    mask={false}
    visible={false}
  />
</div>
`;
